### ISSUE TITLE
`--no-configured-files` should not update `current_version`

### ISSUE THREAD

**Issue by kdeldycke** [callowayproject/bump-my-version#25]
* bump-my-version version: `0.5.1.dev6` from https://github.com/callowayproject/bump-my-version/pull/23
* Python version: `3.11.4`
* Operating System: macOS 13.4

### Description

When invoking the CLI with the `--no-configured-files`, I always ends up with the `current_version` tracked by `bump-my-version` being updated in its configuration file.

Instead, I expect the `current_version` to be left as-is, as the description of `--no-configured-files` seems to imply: `Only replace the version in files specified on the command line`.

### What I Did

Here I try to search and replace strings in the `./.github/workflows/autofix.yaml` file:
```shell-session
$ bump-my-version bump --verbose --no-configured-files --search "/workflows/main/" --replace "/workflows/v{current_version}/" patch ./.github/workflows/autofix.yaml
Starting BumpVersion 0.5.1.dev6                                                                                                                                                                                     
Reading config file pyproject.toml:                                                                                                                                                                                 
Specified version (2.17.5) does not match last tagged version (2.17.4)                                                                                                                                              
Parsing version '2.17.5' using regexp '(?P<major>\d+)\.(?P<minor>\d+)\.(?P<patch>\d+)'                                                                                                                              
Parsed the following values: major=2, minor=17, patch=5                                                                                                                                                             
Attempting to increment part 'patch'                                                                                                                                                                                
Values are now: major=2, minor=17, patch=6                                                                                                                                                                          
New version will be '2.17.6'                                                                                                                                                                                        
Asserting files ./.github/workflows/autofix.yaml contain the version string...                                                                                                                                      
Found '/workflows/main/' in ./.github/workflows/autofix.yaml at line 54:           https://raw.githubusercontent.com/kdeldycke/workflows/main/requirements.txt                                                      
Changing file ./.github/workflows/autofix.yaml:                                                                                                                                                                     
*** before ./.github/workflows/autofix.yaml                                                                                                                                                                         
--- after ./.github/workflows/autofix.yaml                                                                                                                                                                          
***************                                                                                                                                                                                                     
*** 52,65 ****                                                                                                                                                                                                      
        - name: Install Poetry                                                                                                                                                                                      
          run: >                                                                                                                                                                                                    
            python -m pip install --requirement                                                                                                                                                                     
!           https://raw.githubusercontent.com/kdeldycke/workflows/main/requirements.txt                                                                                                                             
        - name: Project metadata                                                                                                                                                                                    
          id: project-metadata                                                                                                                                                                                      
          env:                                                                                                                                                                                                      
            GITHUB_CONTEXT: ${{ toJSON(github) }}                                                                                                                                                                   
          run: >                                                                                                                                                                                                    
            python -c "$(curl -fsSL                                                                                                                                                                                 
!           https://raw.githubusercontent.com/kdeldycke/workflows/main/.github/metadata.py)"                                                                                                                        
                                                                                                                                                                                                                    
    format-python:                                                                                                                                                                                                  
      name: Format Python                                                                                                                                                                                           
--- 52,65 ----                                                                                                                                                                                                      
        - name: Install Poetry                                                                                                                                                                                      
          run: >                                                                                                                                                                                                    
            python -m pip install --requirement                                                                                                                                                                     
!           https://raw.githubusercontent.com/kdeldycke/workflows/v2.17.5/requirements.txt                                                                                                                          
        - name: Project metadata                                                                                                                                                                                    
          id: project-metadata                                                                                                                                                                                      
          env:                                                                                                                                                                                                      
            GITHUB_CONTEXT: ${{ toJSON(github) }}                                                                                                                                                                   
          run: >                                                                                                                                                                                                    
            python -c "$(curl -fsSL                                                                                                                                                                                 
!           https://raw.githubusercontent.com/kdeldycke/workflows/v2.17.5/.github/metadata.py)"                                                                                                                     
                                                                                                                                                                                                                    
    format-python:                                                                                                                                                                                                  
      name: Format Python                                                                                                                                                                                           
***************                                                                                                                                                                                                     
*** 78,84 ****                                                                                                                                                                                                      
        - name: Install ruff, black, blacken-docs, autopep8 and docformatter                                                                                                                                        
          run: >                                                                                                                                                                                                    
            python -m pip install --requirement                                                                                                                                                                     
!           https://raw.githubusercontent.com/kdeldycke/workflows/main/requirements.txt                                                                                                                             
        - name: Run Ruff                                                                                                                                                                                            
          # --select=ALL - Autofix everything.                                                                                                                                                                      
          #   Includes rules in the nursery, a collection of newer lints that are still under development.                                                                                                          
--- 78,84 ----                                                                                                                                                                                                      
        - name: Install ruff, black, blacken-docs, autopep8 and docformatter                                                                                                                                        
          run: >                                                                                                                                                                                                    
            python -m pip install --requirement                                                                                                                                                                     
!           https://raw.githubusercontent.com/kdeldycke/workflows/v2.17.5/requirements.txt                                                                                                                          
        - name: Run Ruff                                                                                                                                                                                            
          # --select=ALL - Autofix everything.                                                                                                                                                                      
          #   Includes rules in the nursery, a collection of newer lints that are still under development.                                                                                                          
***************                                                                                                                                                                                                     
*** 157,163 ****                                                                                                                                                                                                    
        - name: Install mdformat                                                                                                                                                                                    
          run: >                                                                                                                                                                                                    
            python -m pip install --requirement                                                                                                                                                                     
!           https://raw.githubusercontent.com/kdeldycke/workflows/main/requirements.txt                                                                                                                             
        - name: Install shfmt                                                                                                                                                                                       
          run: |                                                                                                                                                                                                    
            sudo apt install --yes shfmt                                                                                                                                                                            
--- 157,163 ----                                                                                                                                                                                                    
        - name: Install mdformat                                                                                                                                                                                    
          run: >                                                                                                                                                                                                    
            python -m pip install --requirement                                                                                                                                                                     
!           https://raw.githubusercontent.com/kdeldycke/workflows/v2.17.5/requirements.txt                                                                                                                          
        - name: Install shfmt                                                                                                                                                                                       
          run: |                                                                                                                                                                                                    
            sudo apt install --yes shfmt                                                                                                                                                                            
Writing to config file pyproject.toml:                                                                                                                                                                              
*** before pyproject.toml                                                                                                                                                                                           
--- after pyproject.toml                                                                                                                                                                                            
***************                                                                                                                                                                                                     
*** 1,5 ****                                                                                                                                                                                                        
  [tool.bumpversion]                                                                                                                                                                                                
! current_version = "2.17.5"                                                                                                                                                                                        
  allow_dirty = true                                                                                                                                                                                                
                                                                                                                                                                                                                    
  [[tool.bumpversion.files]]                                                                                                                                                                                        
--- 1,5 ----                                                                                                                                                                                                        
  [tool.bumpversion]                                                                                                                                                                                                
! current_version = "2.17.6"                                                                                                                                                                                        
  allow_dirty = true                                                                                                                                                                                                
                                                                                                                                                                                                                    
  [[tool.bumpversion.files]]                                                                                                                                                                                        
Would not commit                                                                                                                                                                                                    
Would not tag since we are not committing   
```

This perfectly does the job I'm expecting it to do. 💪

With one exception: I did not expected the `current_version` in my `pyproject.toml` file to be updated.

I have to admit this might be confusing, as I actually provided the `patch` part to the command line. Maybe that was because I am trying to use `bump-my-version` as a kind of search & replace tool, but augmented with its version management? Thinking of it maybe what I'm trying to so is the equivalent of the `bump` subcommand, without the effective increment of the version internally tracked by `bump-my-version`.

So all in all, how should we qualify this use-case?
- as a new option to `bump` subcommand (like `--no-increment`)?
- as a new `replace` subcommand (i.e. same as `bump` without the automatic increment)?
- as a change in behavior for the `--no-configured-files` option?

**Comment by kdeldycke** [comment]
Now thinking of it, maybe this will simply be solved while addressing https://github.com/callowayproject/bump-my-version/issues/19 , or at least be made more clear and explicit by the decoupling of version increment and search & replace.

In the current state of things #23 is not enough to address this use case. I'll revisit this issue at the first hint of progress on #19.

**Comment by kdeldycke** [comment]
Revisiting this issue with 0.7.0, which has the exact same behaviour as reported initially.

But for my particular use case, the new `replace` subcommand while solve this issue for me once #34 is addressed.

**Comment by coordt** [comment]
Version 0.7.1 was just released and fixes #34

Let me know if your issue is fixed and we can close this or more is required.

**Comment by kdeldycke** [comment]
For the record, the new and clean form of invokation to cover my use case is, since 0.7.1:
```shell-session
$ bump-my-version replace --verbose --no-configured-files --search "/workflows/main/" --replace "/workflows/v{current_version}/" ./.github/workflows/autofix.yaml
```
